ocs_ci/ocs/container.py

```diff
@@ -0,0 +1,250 @@
+
+import logging
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.utility import templating
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.exceptions import CommandFailed
+
+from tests.helpers import create_unique_resource_name, \
+    get_admin_key, get_cephfs_name, get_cephfs_data_pool_name
+
+logger = logging.getLogger(__name__)
+
+
+class Container(object):
+    """
+    Container Storage class
+
+    TODO: Document Args
+    """
+    def __init__(self,
+                 interface_type=constants.CEPHBLOCKPOOL,
+                 block_pool_name=None,
+                 reclaim_policy=constants.RECLAIM_POLICY_DELETE,
+                 sc_name=None,
+                 pvc_name=None,
+                 access_mode=constants.ACCESS_MODE_RWO,
+                 namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                 wait=True,
+                 size=None,
+                 cleanup=False,
+                 ):
+        self.interface_type = interface_type
+        self.pool_name = block_pool_name
+        self.sc_name = sc_name
+        self.pvc_name = pvc_name
+        self.access_mode = access_mode
+        self.reclaim_policy = reclaim_policy
+        self.namespace = namespace
+        self.size = size
+        self.wait = wait
+        self.sc_data = dict()
+        self.block_data = dict()
+        self.secret_data = dict()
+        self.pvc_data = dict()
+        if cleanup:
+            self._cleanup()
+        if interface_type == constants.CEPHBLOCKPOOL:
+            self._setup_rbd_secret()
+            self._setup_ceph_blockpool()
+            self._setup_rbd_storageclass()
+        else:
+            self._setup_fs_secret()
+            self._setup_fs_storageclass()
+        self._setup_pvc()
+
+    def _setup_rbd_secret(self):
```

Comment:
would it make more sense to call the functions in helpers.py like create_secret()? I guess this will require moving helpers to a higher dir hierarchy

Reply:
its actually using couple of functions from helpers, for few things like this I need to add this as attribute of object.